ocaml/xapi/xapi_host.ml

```diff
@@ -2928,6 +2928,7 @@ let apply_updates ~__context ~self ~hash =
 
 let set_https_only ~__context ~self ~value =
   let state = match value with true -> "close" | false -> "open" in
-  ignore
-  @@ Helpers.call_script !Xapi_globs.firewall_port_config_script [state; "80"] ;
-  Db.Host.set_https_only ~__context ~self ~value
+  let network_state =
+    Helpers.call_script !Xapi_globs.firewall_port_config_script [state; "80"]
+  in
+  Db.Host.set_https_only ~__context ~self ~value:(bool_of_string network_state)
```

Comment:
This change should not be done here, but in `refresh_localhost_info` in dbsync_slave.ml, which is a function that runs when xapi start up. At that point we want to query the state of the firewall and sync the DB.